tests/simulation/sim/sim_utils.go

```diff
@@ -1,12 +1,16 @@
 package sim
```

Comment:
I don't think we need a directory `tests`, we can directly put `simulation` at the root

I think this can also be removed from test coverage

Reply:
I was planning to have a directory test that can hold all test-related files , and have directories for simulation, e2e etc

Reply:
I would do `./simtest/*` directly in the project's root in a single package to keep it simple. 

- `simtest`
  - `utils.go`
  - `config.go`
  - `state.go`
  - `abc_test.go`
  - `xyz_test.go`
  - `...`

Reply:
I would keep `e2e` at the root

Reply:
I guess both comments are similar in keeping both e2e and simulation under root, 
I would have preferred to have a tests directory ,as it would reduce the total number of folders directly under root, but don't feel strongly about it .
Moved simulation directly under root for now